grafanalib/core.py

```diff
@@ -2692,6 +2713,50 @@ def to_json_data(self):
         }
 
 
+@attr.s
+class GraphThreshold(object):
+    """Threshold for for Graph panel
+    (https://grafana.com/docs/grafana/latest/panels/thresholds/)
+
+    :param colorMode: Color mode of the threshold, value can be `ok`, `warning`, `critical` or `custom`.
+        If `custom` is selcted a lineColor and fillColor should be provided
+    :param fill: Display threshold fill, defaults to True
+    :param line: Display threshold line, defaults to True
+    :param value: When to use this color will be null if index is 0
+    :param op: EVAL_LT for less than or EVAL_GT for greater than to indicate what the threshold applies to.
+    :param yaxis: Choose left or right for Graph panels
+    :param fillColor: Fill color of the threshold, when colorMode = "custom"
+    :param lineColor: Line color of the threshold, when colorMode = "custom"
+
+    Example:
+        thresholds = [
+            GraphThreshold(colorMode="ok", value=10.0),
+            GrpahThreshold(colorMode="critical", value=90.0)
+        ]
+    """
+
+    value = attr.ib(validator=instance_of(float))
+    colorMode = attr.ib(default="critical")
+    fill = attr.ib(default=True, validator=instance_of(bool))
+    line = attr.ib(default=True, validator=instance_of(bool))
+    op = attr.ib(default=EVAL_GT)
+    yaxis = attr.ib(default='left')
+    fillColor = attr.ib(default=RED)
+    lineColor = attr.ib(default=RED)
```

Comment:
`fillColor` and `lineColor` should be defined only for `custom` `colorMode`. I don't think a default value for them should be defined as they are optional.

Reply:
I have made them conditional, but have left the defaults as didnt want to just put empty string